### PR TITLE
fix: status wrong when task processed in the delegation api

### PR DESCRIPTION
**Pull request by zhujian7**
**What this PR does / why we need it**:

If the delegation task processed in the delegation API and update the task status to Succeeded/Failed status, we should not update it to Waiting(since the task has already been processed), otherwise, the task will be stuck.

**Which issue(s) this PR is related to** *(optional, link to 3rd issue(s))*:

Fixes #

Reference to #

**Special notes for your reviewer**:

/cc @hyy0322 @supereagle 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by hyy0322** [comment]
Delegation task is something like calling external service like jenkins to do some work?

**Comment by hyy0322** [comment]
Why deletion task processed twice?

File: pkg/workflow/workflowrun/workload.go [hunk 1]
```diff
@@ -144,6 +144,22 @@ func (p *WorkloadProcessor) processDelegation() error {
...
+
+	// If the task has already been processed by the above RESTful API task, we should not update the status of the task to Waiting
+	latestWfr, err := p.client.CycloneV1alpha1().WorkflowRuns(p.wfr.Namespace).Get(p.wfr.Name, metav1.GetOptions{})
+	if err != nil {
+		log.WithField("wfr", p.wfr.Name).Error("Get latest wfr failed")
+		return err
+	}
+	// The task maybe has been processed by the delegation task and status not needed to be set to Waiting
+	if stg, ok := latestWfr.Status.Stages[p.stg.Name]; ok {
+		if stg.Status.Phase == v1alpha1.StatusSucceeded ||
```
Comment (supereagle) [inline pkg/workflow/workflowrun/workload.go:156]: I remember that we have defined a method for this status judgement. If not, I think we should define one for this.

**Comment by supereagle** [comment]
ping @zhujian7

File: pkg/workflow/workflowrun/workload.go (hunk 1 above)
Comment (zhujian7) [reply pkg/workflow/workflowrun/workload.go]: OK, I will refactor it.

**Comment by zhujian7** [comment]
> ping @zhujian7

@supereagle  PTAL

**Pull request merged**